client/VIEW_MAP_DATA.md

```diff
@@ -0,0 +1,177 @@
+
+When the map is clicked, the data (aka features - which are sets of properties in the geojson file ) of the map are able to be accessed via the MapEvent. See [j40Map.tsx](https://github.com/usds/justice40-tool/blob/main/client/src/components/J40Map.tsx)'s `onClick` method for more details.
+
+## Troubleshooting
+
+If the map is not loading. There are some things it could be
+
+- [Browser caching](#browser-caching)
+- [S3 Permissions](#s3-permissions)
+- [CDN caching](#cdn-caching)
+- [Geojson contract failure](#geojson-contract-failure)
+
+### Browser caching
+
+Disable caching in the browser and clear the cache.
+
+### S3 Permissions
+
+When the tile data is updated, it's possible that the permissions on the S3 bucket may be reset to be non-publically accessible. The CDN mentioned above points to an underlying S3 bucket. Check the permissions by curling the underlying S3 bucket directly and inspecting the headers:
+
+`curl -D - http://justice40-data.s3.amazonaws.com/data-pipeline/data/score/tiles/low/3/1/3.pbf`
+
+If permissions are not set up correctly the underlying S3 location will not be accessible (will show a 403 via curl) or a permission XML error when accessing via the browser. This curl is done to check that the file is public, if not, there could be an issue with the map reading it. It should return 200 with a reasonble file size on the order of KB.
+
+Note also that the browser will also need to run the localhost on http as well.
+Update permissions in s3 and try again.
+
+### CDN caching
+
+When the S3 data is updated, it may not invalidate the cache in the CDN. You may have to invalidate the cache. There should be some existing invalidations in place to copy when having to invalidate.
+
+#### Set app to hit S3
+
+You can change the CDN's URL to directly go to S3. This will require going over non-ssl connection. Set your localhost to use a non-ssl protocol (http rather than https). This another way to remove the CDN to help troubleshoot.
+
+This invalidation should be done by a GHA when the tiles are updating.
+
+When in prod, the first thing is to check the s3 access vs CDN.
+
+### Geojson contract failure
+
+The frontend map (via the [constants](https://github.com/usds/justice40-tool/blob/main/client/src/data/constants.tsx) file) expects that certain properties in the geojson file:
+
+1. exist and
+2. have certain values
+
+If these properties are missing or have different values from what's in the constants file, the map may not show the data layer.
+
+#### Geojson data contract
+
+Below we will show some sample geojson files. The map currently requires different data for the high zoom and low zoom levels.
+
+The geojson shown below is for the low zoom values (when the map is first opened, it starts at zoom level 3).
+
+```json
```

Comment:
To clarify how to use the `constants` file and how that relates to what's in the JSON, how about adding comments inline with this JSON example naming the constant that corresponds to the given field?